### ISSUE TITLE
[Google Hotels API] Searches return empty results even thought results are available via regular search

### ISSUE THREAD

**Issue by schaferyan** [serpapi/public-roadmap#2101]
An Enterprise Customer reported that some of their Google Hotels API searches return an empty response, even though they should have results.

<img width="1197" alt="Screenshot 2024-11-06 at 9 56 11 AM" src="https://github.com/user-attachments/assets/f9bf5404-d3ba-4b0d-9a9f-96cd1494e806">

The customer shared an example, and I've asked for more examples if possible. I was unable to replicate it in the Playground, despite numerous retries. 

Public links: [Playground](https://serpapi.com/playground?engine=google_hotels&q=berlin&gl=us&hl=en&currency=USD&check_in_date=2024-11-18&check_out_date=2024-11-19&adults=2&no_cache=true)

Internal links:[ Front](https://app.frontapp.com/open/msg_2jnsgvx8?key=YFrPqPIAIab7DJJI7Hw-AeYva5DZV6KY) | [Search Inspect ](https://serpapi.com/searches/6720f70dc82127abaccdbb6e/inspect)

**Comment by schaferyan** [comment]
The enterprise customer who reported this followed up with some more examples:

[Front](https://app.frontapp.com/open/msg_2k4lm7fg?key=Z9A8Sow_id1Nc08Js_B8oq7LgwV5kkTj)

**Comment by sanj-directbooker** [comment]
What's the endpoint that the Hotels request is running against?  I'd like to know if it is different than the one being used for Maps.

**Comment by sanj-directbooker** [comment]
I have an idea of what might be happening.

The autocomplete mechanism that Google uses defaults to returning a full (over-qualified) name for what the user enters.

You need to explicitly specify that you want the "name" field: https://developers.google.com/maps/documentation/javascript/reference/places-service#PlaceResult.name

This is the field that the Google Hotels input field expects.

As an example, Google Hotels fails on the fully qualified name for the Berlin hotel (Dorint Kurfürstendamm Berlin, Augsburger Straße, Berlin-Charlottenburg-Wilmersdorf, Germany) but succeeds on the shorter name (Dorint Kurfürstendamm Berlin).  This is the name shown as part of the hotel's display in Google: https://www.google.com/travel/search?q=Dorint%20Kurf%C3%BCrstendamm%20Berlin&qs=CAAgACgA&ved=0CAAQ5JsGahgKEwiA29H71OuJAxUAAAAAHQAAAAAQmwI&ts=CAEqBwoFOgNVU0Q&ap=KigKEgnuQR5ZFEBKQBGp-gzjIKkqQBISCbeKF8-eQEpAEan6DCPBqipAMAA

The same thing happens with my other example: the fully overqualified name fails on Google Hotels (H10 Port Vell, Pas de Sota Muralla, Barcelona, Spain), but the short name works (H10 Port Vell).

NOTE:  This is an issue for individual hotel names not for cities.  For individual hotels, Google Hotels expects the user to select an entry from the list of autocomplete elements, not simply hit return.

**Comment by sanj-directbooker** [comment]
tl;dr: Google Hotels is expecting a selection from its autocomplete set whereas you're given it a search term.  This works for cities, but not for individual hotels.

**Comment by schaferyan** [comment]
Thanks for the additional information @sanj-directbooker! I've opened an issue to support scraping Google Hotels Autocomplete suggestions. This could include support for extracting an ID from each autocomplete suggestion, then using that to query the Google Hotels API.

You can find that issue here:

https://github.com/serpapi/public-roadmap/issues/2159

**Comment by sanj-directbooker** [comment]
any progress on this?

**Comment by tanys123** [comment]
Hi @sanj-directbooker, we have deployed a potential fixed for the issue. Please try it out and let us know if that works for you.

<img width="1366" alt="Screenshot 2024-12-11 at 10 07 31 AM" src="https://github.com/user-attachments/assets/575993f0-38a9-4a04-b00a-73b82c766c87">

**Comment by sanj-directbooker** [comment]
This works!  thanks!

**Comment by sanj-directbooker** [comment]
Hey there.  I tested this more and it still doesn't work correctly.

This search works:  https://serpapi.com/playground?engine=google_hotels&q=H10+Port+Vell&gl=us&hl=en&currency=USD&check_in_date=2025-01-07&check_out_date=2025-01-08

But this one still doesn't: https://serpapi.com/playground?engine=google_hotels&q=H10+Port+Vell%2C+Pas+de+Sota+Muralla%2C+Barcelona%2C+Spain&gl=us&hl=en&currency=USD&check_in_date=2025-01-07&check_out_date=2025-01-08

**Comment by schaferyan** [comment]
Hi @sanj-directbooker, thanks for your feedback. Can you share a link to the search you are comparing this to in your browser?

When I try this search directly in Google, it doesn't return any results:

https://www.google.com/travel/search?q=hotelsH10%20Port%20Vell%2C%20Pas%20de%20Sota%20Muralla%2C%20Barcelona%2C%20Spain&g2lb=4965990%2C4969803%2C72277293%2C72302247%2C72317059%2C72406588%2C72414906%2C72421566%2C72471280%2C72472051%2C72481459%2C72485658%2C72499705%2C72560029%2C72573224%2C72614662%2C72616120%2C72619927%2C72628720%2C72647020%2C72648289%2C72658035%2C72686036%2C72760082%2C72803964%2C72808078%2C72832973&hl=en-US&gl=us&ssta=1&ts=CAESCgoCCAMKAggDEAAqCQoFOgNVU0QaAA&qs=CAAgACgA&ap=MABoAQ&ictx=111&ved=0CAAQ5JsGahgKEwiooM-bjuKKAxUAAAAAHQAAAAAQuQI

**Comment by schaferyan** [comment]
Hi @sanj-directbooker, thanks for your feedback!

I'm not getting any results for that query when I try it via Google Hotels directly with the browser in incognito mode:

https://www.google.com/travel/search?q=H10%20Port%20Vell%2C%20Pas%20de%20Sota%20Muralla%2C%20Barcelona%2C%20Spain&g2lb=4965990%2C4969803%2C72277293%2C72302247%2C72317059%2C72406588%2C72414906%2C72421566%2C72471280%2C72472051%2C72481459%2C72485658%2C72499705%2C72560029%2C72573224%2C72614662%2C72616120%2C72619927%2C72647020%2C72648289%2C72658035%2C72686036%2C72760082%2C72803964%2C72808078%2C72832973&hl=en-US&gl=us&ssta=1&ts=CAEaIAoCGgASGhIUCgcI6Q8QARgHEgcI6Q8QARgIGAEyAggBKgQKABoA&qs=CAEgACgASAA&ap=MAFoAQ&ictx=111&ved=0CAAQ8IAIahgKEwiYuNO1j-KKAxUAAAAAHQAAAAAQ_AM

If you're getting results that way on your side, can you please share the URL?

**Comment by sanj-directbooker** [comment]
I don't appear to be getting responses from Google either.  I'm coming to the conclusion that the Google autocomplete API doesn't work well with Hotel Search.

**Comment by martin-serpapi** [comment]
Hi @sanj-directbooker , thank you for getting back to us. Feel free to contact us if you notice any other issues, or if you have any questions or concerns. We'll be happy to help.

**Comment by sanj-directbooker** [comment]
One thing I just noticed: when the "single property" response comes back, it doesn't include the property_token.

compare this: https://serpapi.com/playground?engine=google_hotels&q=head+of+the+river%2C+Oxford&gl=us&hl=en&currency=USD&check_in_date=2025-01-12&check_out_date=2025-01-14&adults=2&newPara=location

to the fields here: https://serpapi.com/playground?engine=google_hotels&q=Oxford%2C+uk&gl=us&hl=en&currency=USD&check_in_date=2025-01-12&check_out_date=2025-01-14&adults=2&newPara=location

What's missing in the first case are these two fields:

"property_token": "ChgIx_aNxN2BrL7RARoLL2cvMXRkcjJ0Y20QAQ",
"serpapi_property_details_link": "https://serpapi.com/search.json?adults=2&check_in_date=2025-01-12&check_out_date=2025-01-14&children=0¤cy=USD&engine=google_hotels&gl=us&hl=en&property_token=ChgIx_aNxN2BrL7RARoLL2cvMXRkcjJ0Y20QAQ&q=Oxford%2C+uk"

**Comment by sanj-directbooker** [comment]
It'd be super useful for us.  It lets us easily keep the list to detail
requests the same as repeat detail requests.

Adding that field would be very much appreciated!

sanjay

**Comment by martin-serpapi** [comment]
Thank you for the confirmation, @sanj-directbooker . I opened a feature request for this. You can follow our progress and add any comments here - #2300
